catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -13,6 +13,1240 @@
+        when(security.runWithSubjectOrElevate(any(Callable.class))).thenAnswer(invocation -> {
+            try {
+                return ((Callable) invocation.getArguments()[0]).call();
+            } catch (Exception e) {
+                throw new InvocationTargetException(e.getCause());
+            }
+        });
+
+    }
+
+    @Test
+    public void testGetRegistryObjectByMetacardId() throws Exception {
+        String metacardId = TEST_METACARD_ID;
+        Metacard metacard = testMetacard;
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request, metacard);
+        JAXBElement<RegistryPackageType> jaxbRegistryPackage =
+                EbrimConstants.RIM_FACTORY.createRegistryPackage(getTestRegistryPackage());
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        when(parser.unmarshal(any(ParserConfigurator.class),
+                eq(JAXBElement.class),
+                any(InputStream.class))).thenReturn(jaxbRegistryPackage);
+        RegistryPackageType registryPackage =
+                federationAdminServiceImpl.getRegistryObjectByRegistryId(metacardId, null);
+        assertThat(registryPackage, notNullValue());
+        verify(federationAdminServiceImpl).getRegistryObjectByRegistryId(metacardId, null);
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void getRegistryObjectByRegistryIdEmptyString() throws FederationAdminException {
+        assertThat(federationAdminServiceImpl.getRegistryObjectByRegistryId(""), is(nullValue()));
+    }
+
+    @Test
+    public void getRegistryObjectByRegistryIdThatDoesNotExist() throws Exception {
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request);
+        Metacard metacard = getTestMetacard();
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        when(registryTransformer.transform(any(InputStream.class))).thenReturn(metacard);
+        assertThat(federationAdminServiceImpl.getRegistryObjectByRegistryId("Not a metacard"),
+                is(nullValue()));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void getRegistryObjectByRegistryIdDuplicateMetacards() throws Exception {
+        String metacardId = TEST_METACARD_ID;
+        Metacard metacard = testMetacard;
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request, metacard, metacard);
+        when(security.getSystemSubject()).thenReturn(subject);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        federationAdminServiceImpl.getRegistryObjectByRegistryId(metacardId);
+        verify(catalogFramework).query(any(QueryRequest.class));
```

Comment:
We should do more validation on the `QueryRequest` object. If we can't just use simple equality, consider using http://site.mockito.org/mockito/docs/current/org/mockito/ArgumentCaptor.html.